### ISSUE TITLE
Cannot authenticate in Printers Troubleshoot wizard

### ISSUE THREAD

**Issue by bshanks** [pop-os/pop#1262]
Distribution (run cat /etc/os-release):
NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os

Related Application and/or Package Version (run apt policy $PACKAGE NAME):
apt policy cups
cups:
Installed: 2.3.1-9ubuntu1.1
Candidate: 2.3.1-9ubuntu1.1
Version table:
*** 2.3.1-9ubuntu1.1 500
500 http://us.archive.ubuntu.com/ubuntu focal-security/main amd64 Packages
500 http://us.archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages
100 /var/lib/dpkg/status
2.3.1-9ubuntu1 500
500 http://us.archive.ubuntu.com/ubuntu focal/main amd64 Packages

**Issue/Bug Description:**
Printer troubleshooter causes error: "Not authorized" in GUI

**Steps to reproduce (if you know):**
Settings->Printers->Additional printer settings...->Help->Troubleshoot
Go thru the wizard. After the Enable Debugging screen appears, an authentication dialog box pops up. No known username/password combination works. 

/var/log/auth.log contains entries such as:
Sep 29 10:50:00 pop-os cupsd: pam_unix(cups:auth): Couldn't open /etc/securetty: No such file or directory
Sep 29 10:50:00 pop-os cupsd: pam_unix(cups:auth): Couldn't open /etc/securetty: No such file or directory
Sep 29 10:50:00 pop-os cupsd: pam_unix(cups:auth): authentication failure; logname= uid=0 euid=0 tty=cups ruser= rhost=localhost  user=root

**Expected behavior:**
If authentication is needed to debug printing, then either the user's password should work, or the same password used by the 'sudo' command should work. Also, the authentication dialog box should be preceded by some guidance about why authentication is being requested, and which username should be entered.

**Other Notes:**

**Comment by jacobgkau** [comment]
The password in the GUI dialog should be your own user's password, which is also what the `sudo` command will ask for by default. (There is no separate root password by default.) Can you please make sure your user is part of the `lpadmin` group using this command:

```
sudo usermod -aG lpadmin $USER
```

After running that command and rebooting, are you able to authenticate in that dialog?

**Comment by bshanks** [comment]
Thank you!

I think it's likely that my user not being in 'lpadmin' was indeed the problem there. Overnight I managed to fix the printer setup, and the Troubleshooting wizard is going down a different path now, so I can't directly verify, however one of the things I did while fixing it was add my user to lpadmin, so that probably did help. I suggest having pop-os add the primary user account to lpadmin upon creation.

Related, when fixing the issue with my printer I encountered further issues:
- Using the big 'add a printer' button in the Settings dialog box did locate my printer and attempt to set it up, however printing to it would either cause a message saying that the job was unable to be printed, or else the print job showed as completed (even in the print queue) but the printer didn't actually print anything, depending on which application was used to print.
- One problem was that the printer driver ("make/model") that was automagically selected was Generic/text, when in fact the correct printer driver turned out to be Generic/IPP everywhere
- Another problem was that, even if I manually selected Generic/IPP everywhere during printer creation, it didn't work. I ended up having to do "sudo vi /etc/cups/cupsd.conf", add the line "Listen localhost:631", and then did 'service cups restart', in order to enable the CUPS web interface on port 631. For some reason, adding the printer on the web interface with Generic/IPP everywhere did work (there was a small difference that I noticed; the printer settings window asked me if the printer had a duplexer, and I answered Yes; the CUPS web interface didn't ask me that question. I did not go back later to re-add the printer while saying No to the duplexer question to see if it made a difference).
Imo think the fact that adding the printer through the printer settings windows has a different result than adding it through the CUPS web interface is an issue in itself. Furthermore, if the CUPS web interface works better than the printer settings windows, then perhaps it should be on by default, although perhaps that causes security issues in which case I understand why it is off.
